main.tf

```
@@ -1 +1,193 @@
-# Your Terraform code goes here :)
+locals {
+  access_token_json = var.use_ssm_parameter_store ? (
+    {
+      "secrets" : [{
+        "name" : "LaceworkAccessToken",
+        "valueFrom" : local.ssm_parameter_arn
+      }]
+    }
+    ) : (
+    {
+      "environment" : [{
+        "name" : "LaceworkAccessToken",
+        "value" : var.lacework_access_token
+      }]
+    }
+  )
+  container_definition_json = jsonencode([merge(
+    local.access_token_json,
+    {
+      "essential" : true,
+      "image" : "lacework/datacollector",
+      "mountPoints" : [
+        {
+          "readOnly" : true,
+          "containerPath" : "/laceworkfim",
+          "sourceVolume" : "root"
+        },
+        {
+          "readOnly" : null,
+          "containerPath" : "/var/lib/lacework",
+          "sourceVolume" : "var_lib_lacework"
+        },
+        {
+          "readOnly" : null,
+          "containerPath" : "/var/log",
+          "sourceVolume" : "var_log"
+        },
+        {
+          "readOnly" : null,
+          "containerPath" : "/var/run",
+          "sourceVolume" : "var_run"
+        },
+        {
+          "readOnly" : true,
+          "containerPath" : "/etc/passwd",
+          "sourceVolume" : "etc_passwd"
+        },
+        {
+          "readOnly" : true,
+          "containerPath" : "/etc/group",
+          "sourceVolume" : "etc_group"
+        }
+      ],
+      "name" : local.ecs_task_family_name,
+      "privileged" : true
+    }
+  )])
+  ecs_service_name     = length(var.ecs_service_name) > 0 ? var.ecs_service_name : "${var.resource_prefix}-service-${random_id.uniq.hex}"
+  ecs_task_family_name = length(var.ecs_task_family_name) > 0 ? var.ecs_task_family_name : "${var.resource_prefix}-task-${random_id.uniq.hex}"
+  iam_role_arn         = var.use_existing_iam_role ? var.iam_role_arn : aws_iam_role.ecs_execution[0].arn
+  iam_role_name        = (!var.use_existing_iam_role && length(var.iam_role_name) > 0) ? var.iam_role_name : "${var.resource_prefix}-role-${random_id.uniq.hex}"
+  ssm_parameter_arn    = (var.use_ssm_parameter_store && !var.use_existing_ssm_parameter) ? aws_ssm_parameter.lacework_access_token[0].arn : var.ssm_parameter_arn
+}
+
+resource "random_id" "uniq" {
+  byte_length = 4
+}
+
+resource "aws_iam_role" "ecs_execution" {
+  count = var.use_existing_iam_role ? 0 : 1
+
+  name = local.iam_role_name
+  tags = var.iam_role_tags
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2008-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "ecs_task_execution_policy_attachment" {
+  count = var.use_existing_iam_role ? 0 : 1
+
+  role       = aws_iam_role.ecs_execution[0].name
```

I think it is better to do this:
```suggestion
  role       = aws_iam_role.ecs_execution[count.index].name
```